### PR TITLE
Include competition rules

### PR DESCRIPTION
**Pull request by fsschneider**
This PR adds the Competition Rules, moves what was previously called the rules to a technical documentation page, adds the Call for Submissions, and makes a few rules changes as discussed in the WG meetings.

@georgedahl I want to highlight the following changes in particular, that could use your attention:

- The [FAQs](https://github.com/fsschneider/algorithmic-efficiency/blob/dev/DOCUMENTATION.md#faqs) are currently part of the "(Technical) Documentation" doc. Should it rather be a separate file? @priyakasimbeg?
  - I added the question "Can I submit multiple times to the benchmark competition?" with the answer that was originally in the "Submission Process Rules" document. Is [this answer](https://github.com/fsschneider/algorithmic-efficiency/blob/dev/DOCUMENTATION.md#can-i-submit-multiple-times-to-the-benchmark-competition) okay?
  - How can we better phrase the question and answer about ["submitting existing algorithms"](https://github.com/fsschneider/algorithmic-efficiency/blob/dev/DOCUMENTATION.md#can-i-submit-existing-ie-published-training-algorithms-as-submissions)?
- I removed mentions of the spirit jury in the "Technical Documentation" doc, see [this commit](https://github.com/mlcommons/algorithmic-efficiency/commit/8b14658ec3eb4ecb0fe6eec931b334fa75b2e1f2). Any edits you'd like to see (in particular to [this section](https://github.com/fsschneider/algorithmic-efficiency/blob/dev/DOCUMENTATION.md#valid-submissions))
- I clarified that we do tuning also for the held-out workloads, see [this commit](https://github.com/mlcommons/algorithmic-efficiency/commit/da11c6766c119ad248eb03d0530cbbed3e342a4d). Is the text okay?
- I added also a [Submission (Google) Form](https://docs.google.com/forms/d/e/1FAIpQLSecR5_5hKolJFJFOZBaf0SRB5CZ1vrYgbxKlOQG-DOo5a5x5w/viewform) (similar to the Intent Registration Form) to the [Competition Rules](https://github.com/fsschneider/algorithmic-efficiency/blob/dev/COMPETITION_RULES.md#how-to-enter). It is currently just a placeholder, saying that the submission portal is currently not open.
- I entirely removed the "Submission Process Rules" as most of it is contained within the "Competition Rules" now. A few notable exceptions are noted below. Is that okay with you that those got removed?
  - There is no statement anymore that members of the awards committee are ineligible for prize money.
  - There is no deadline for a version freeze anymore. Instead, there is now [this section](https://github.com/fsschneider/algorithmic-efficiency/blob/dev/DOCUMENTATION.md#version-freeze) on "Version freeze" in the "Technical Documentation" as well as the first sentence of the ["Cancellation and Modification" Section](https://github.com/fsschneider/algorithmic-efficiency/blob/dev/COMPETITION_RULES.md#cancellation-and-modification) in the "Competition Rules".

**To-dos before merge**

- [x] Specify the threshold submissions (for both rulesets) whose score needs to be exceeded for prize money.
- [x] Get approvable by Peter and David
- [x] Double-check with the lawyers
- [x] Update deadline dates

### CODE REVIEW

**Review by priyakasimbeg — DISMISSED** [review]
LGTM.
I think we can leave the FAQs in the DOCUMENTATION or README because it will be more likely people will notice the section. Alternatively if you think it is too large we can make a separate file and link it from the DOCUMENTATION or README? Either way is fine with me.

**Review by georgedahl — DISMISSED** [review]
Can you add a review comment to any part you think I still need to look at?

File: DOCUMENTATION.md [hunk 1]
```diff
@@ -482,10 +502,81 @@ However, we believe that they are fairer and well-supported by research in machi
...
+
+#### What can I do if running the benchmark is too expensive for me?
+
+Submitters unable to self-fund scoring costs can instead self-report only on the [qualification set of workloads](/COMPETITION_RULES.md#qualification-set) that excludes some of the most expensive workloads. Based on this performance on the qualification set, the working group will provide - as funding allows - compute to evaluate and score the most promising submissions. Additionally, we encourage researchers to reach out to the [working group](mailto:algorithms@mlcommons.org) to find potential collaborators with the resources to run larger, more comprehensive experiments for both developing and scoring submissions.
+
+#### Can I submit existing (i.e. published) training algorithms as submissions?
+
+Yes you may, as long as it isn't an exact copy of an existing submission.
+For example, you may submit the Adam optimizer with your hyperparameters or hyperparameter search spaces, as this constitues a different training algorithm.
```
Comment (fsschneider) [inline DOCUMENTATION.md:569]: @georgedahl Can you double check the answer to this question? I think its phrasing could be improved.

File: DOCUMENTATION.md [hunk 2]
```diff
@@ -237,9 +255,7 @@ Submissions are eligible for an untimed eval every `eval_period` seconds, run as
...
-
-We want to state clearly that we welcome creative ideas and novel research. Therefore, the API aims to allow a wide variety of submissions, however, in some cases, routines that would be allowed in principle might not be practically feasible in the provided framework. The spirit jury, however, will only be invoked for submissions that aim to bypass the core premise of this benchmark since submissions like this would also be irrelevant in practice.
+Unfortunately, we can't easily write rules that make it completely clear if a submission is circumventing the spirit of the benchmark in a way that would encompass all possible cases. We welcome creative ideas and novel research. Therefore, the API aims to allow a wide variety of submissions. However, in some cases, routines that would be allowed in principle might not be practically feasible to express in the provided framework.
```
Comment (fsschneider) [inline DOCUMENTATION.md:258]: @georgedahl I updated the text according to your suggestion but left the first sentence. Without the "we can't easily write rules" it feels weird to have an entire section about allowed and disallowed submissions.

File: DOCUMENTATION.md (hunk 1 above)
Comment (georgedahl) [reply DOCUMENTATION.md:569]: How about:
"Yes you may, as long as it isn't an exact copy of an existing submission.
For example, you may submit the Adam optimizer with your particularly effective hyperparameter search space and hyperparameter configuration, as different choices for hyperparameter values and/or search spaces constitute different training algorithms and are potential sources of innovation." 

If you want, we could also add a bit more text along the following lines (feel free to not add it, but if you do fill in the link to the other question):

"That said, while submitting Adam with some novel heuristic to set various hyperparameters, some especially effective hyperparameter search space, or your single best hyperparameter configuration is fine, avoid making *multiple* submissions that only differ by their hyperparameter configuration without a convincing justification they are substantially different (see "Can I submit multiple times to the benchmark competition?", above)."

**Review by georgedahl — APPROVED** [review]

**Pull request merged**